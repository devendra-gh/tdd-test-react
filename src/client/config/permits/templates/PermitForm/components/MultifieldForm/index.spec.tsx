import React from 'react';
import { shallow } from 'enzyme';
import MultiFieldForm from './index';

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Multifield Form', () => {
  let props: any;
  const isExtraField = false;
  const fieldValue = {};

  beforeEach(() => {
    props = {
      i18n: jest.fn(),
      inputGroups: [
        {
          name: 'title.permitLocations',
          stateKey: 'permitLocations',
          customComponent: 'MultifieldForm',
          hasIncrementButton: false,
          incrementButtonLabel: 'Add Machines',
          fields: [
            {
              name: 'PermitLocation',
              label: 'Permit Location',
              isExtraField,
              value: fieldValue,
              subFields: [
                {
                  'aria-label': 'input number',
                  elementType: 'inputNumber',
                  label: 'paperAd.fieldPeriod',
                  name: 'period',
                  min: 1,
                  max: 12,
                  defaultValue: 1,
                },
                {
                  'aria-label': 'input number',
                  elementType: 'inputNumber',
                  label: 'paperAd.fieldQuantity',
                  name: 'quantity',
                  min: 1,
                  defaultValue: 1,
                },
                {
                  'aria-label': 'badge',
                  bgColor: '#FFF',
                  computedProps: [
                    {
                      name: 'label',
                      dependencies: ['period', 'quantity'],
                      compute: (period = 0, quantity = 0) =>
                        `Total: ${Number(period) * Number(quantity)}`,
                    },
                  ],
                  elementType: 'badge',
                  name: 'total',
                  labelColor: '#161138',
                  items: [
                    {
                      id: 'Autumn Sale',
                      label: 'seasonalPromotions.promotionType.autumnSale',
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    };
  });

  it('should successfully render', () => {
    const component = shallow(<MultiFieldForm {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render fields with computed props', () => {
    props.inputGroups[0].fields[0].isExtraField = true;
    props.inputGroups[0].value = {
      period: '1',
      quantity: '3',
    };

    const component = shallow(<MultiFieldForm {...props} />);
    expect(component).toMatchSnapshot();
  });
});
