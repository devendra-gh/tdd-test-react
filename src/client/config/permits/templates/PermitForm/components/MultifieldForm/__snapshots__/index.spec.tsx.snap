// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multifield Form should render fields with computed props 1`] = `
<form
  name="permitLocations"
>
  <p />
  <div
    style={
      Object {
        "display": "flex",
        "marginBottom": 40,
      }
    }
  >
    <div
      style={
        Object {
          "flexGrow": 15,
        }
      }
    >
      <LocalizedComponent(FormTemplate)
        cancelButton={null}
        cancelLink={null}
        formData={Object {}}
        i18n={[Function]}
        id={null}
        inputGroups={
          Array [
            Object {
              "fields": Array [
                Object {
                  "aria-label": "input number",
                  "defaultValue": 1,
                  "disabledDate": [Function],
                  "elementType": "inputNumber",
                  "key": "period",
                  "label": undefined,
                  "max": 12,
                  "min": 1,
                  "name": "period",
                  "onChange": [Function],
                  "validate": [Function],
                  "value": undefined,
                },
                Object {
                  "aria-label": "input number",
                  "defaultValue": 1,
                  "disabledDate": [Function],
                  "elementType": "inputNumber",
                  "key": "quantity",
                  "label": undefined,
                  "min": 1,
                  "name": "quantity",
                  "onChange": [Function],
                  "validate": [Function],
                  "value": undefined,
                },
                Object {
                  "aria-label": "badge",
                  "bgColor": "#FFF",
                  "computedProps": Array [
                    Object {
                      "compute": [Function],
                      "dependencies": Array [
                        "period",
                        "quantity",
                      ],
                      "name": "label",
                    },
                  ],
                  "defaultValue": null,
                  "disabledDate": [Function],
                  "elementType": "badge",
                  "items": Array [
                    Object {
                      "id": "Autumn Sale",
                      "label": undefined,
                    },
                  ],
                  "key": "total",
                  "label": "Total: 0",
                  "labelColor": "#161138",
                  "name": "total",
                  "onChange": [Function],
                  "validate": [Function],
                  "value": undefined,
                },
              ],
              "isExtraField": true,
              "label": "Permit Location",
              "name": "",
              "subFields": Array [
                Object {
                  "aria-label": "input number",
                  "defaultValue": 1,
                  "elementType": "inputNumber",
                  "label": "paperAd.fieldPeriod",
                  "max": 12,
                  "min": 1,
                  "name": "period",
                },
                Object {
                  "aria-label": "input number",
                  "defaultValue": 1,
                  "elementType": "inputNumber",
                  "label": "paperAd.fieldQuantity",
                  "min": 1,
                  "name": "quantity",
                },
                Object {
                  "aria-label": "badge",
                  "bgColor": "#FFF",
                  "computedProps": Array [
                    Object {
                      "compute": [Function],
                      "dependencies": Array [
                        "period",
                        "quantity",
                      ],
                      "name": "label",
                    },
                  ],
                  "elementType": "badge",
                  "items": Array [
                    Object {
                      "id": "Autumn Sale",
                      "label": "seasonalPromotions.promotionType.autumnSale",
                    },
                  ],
                  "labelColor": "#161138",
                  "name": "total",
                },
              ],
              "value": Object {},
            },
          ]
        }
        name={null}
        submitButton={null}
      />
    </div>
  </div>
  <div
    className="marginT40"
  />
</form>
`;

exports[`Multifield Form should successfully render 1`] = `
<form
  name="permitLocations"
>
  <p />
  <div
    style={
      Object {
        "display": "flex",
        "marginBottom": 40,
      }
    }
  >
    <div
      style={
        Object {
          "flexGrow": 15,
        }
      }
    >
      <LocalizedComponent(FormTemplate)
        cancelButton={null}
        cancelLink={null}
        formData={Object {}}
        i18n={[Function]}
        id={null}
        inputGroups={
          Array [
            Object {
              "fields": Array [
                Object {
                  "aria-label": "input number",
                  "defaultValue": 1,
                  "disabledDate": [Function],
                  "elementType": "inputNumber",
                  "key": "period",
                  "label": undefined,
                  "max": 12,
                  "min": 1,
                  "name": "period",
                  "onChange": [Function],
                  "validate": [Function],
                  "value": undefined,
                },
                Object {
                  "aria-label": "input number",
                  "defaultValue": 1,
                  "disabledDate": [Function],
                  "elementType": "inputNumber",
                  "key": "quantity",
                  "label": undefined,
                  "min": 1,
                  "name": "quantity",
                  "onChange": [Function],
                  "validate": [Function],
                  "value": undefined,
                },
                Object {
                  "aria-label": "badge",
                  "bgColor": "#FFF",
                  "computedProps": Array [
                    Object {
                      "compute": [Function],
                      "dependencies": Array [
                        "period",
                        "quantity",
                      ],
                      "name": "label",
                    },
                  ],
                  "defaultValue": null,
                  "disabledDate": [Function],
                  "elementType": "badge",
                  "items": Array [
                    Object {
                      "id": "Autumn Sale",
                      "label": undefined,
                    },
                  ],
                  "key": "total",
                  "label": "Total: 0",
                  "labelColor": "#161138",
                  "name": "total",
                  "onChange": [Function],
                  "validate": [Function],
                  "value": undefined,
                },
              ],
              "isExtraField": false,
              "label": "Permit Location",
              "name": "",
              "subFields": Array [
                Object {
                  "aria-label": "input number",
                  "defaultValue": 1,
                  "elementType": "inputNumber",
                  "label": "paperAd.fieldPeriod",
                  "max": 12,
                  "min": 1,
                  "name": "period",
                },
                Object {
                  "aria-label": "input number",
                  "defaultValue": 1,
                  "elementType": "inputNumber",
                  "label": "paperAd.fieldQuantity",
                  "min": 1,
                  "name": "quantity",
                },
                Object {
                  "aria-label": "badge",
                  "bgColor": "#FFF",
                  "computedProps": Array [
                    Object {
                      "compute": [Function],
                      "dependencies": Array [
                        "period",
                        "quantity",
                      ],
                      "name": "label",
                    },
                  ],
                  "elementType": "badge",
                  "items": Array [
                    Object {
                      "id": "Autumn Sale",
                      "label": "seasonalPromotions.promotionType.autumnSale",
                    },
                  ],
                  "labelColor": "#161138",
                  "name": "total",
                },
              ],
              "value": Object {},
            },
          ]
        }
        name={null}
        submitButton={null}
      />
    </div>
  </div>
  <div
    className="marginT40"
  />
</form>
`;
